### DISCUSSION TITLE
3.2 TFT  SPI 240x320V1.0 with ILI9341

### DISCUSSION THREAD

**Discussion by georgjorge** [dl9rdz/rdz_ttgo_sonde#438] [TTGO firmware]
Now would like to change to my big display. Unfortunately, neither the wiki proposal nor the way PA2J to connect the display works. Maybe this is an issue of the processor, too.
Did anybodody try this configuration and has an idea to make it working?

**Comment by lenalotti** [comment]
Hello, I am using an 2,8" TFT with the ILI9341 and same resolution as you without any problems on a simple ESP32 DEVKIT V1 board running (version master_v0.9.4). My Hardware configuration is: 
Display type=3; SDA=4; CLK=13; RST=14; RS=2; CS=0; TFT SPI speed=20000000. 
Using other pin numbers should not matter, but the ILI9341 chip is made for a max speed of 20000000, not more !
good luck

**Reply by wrevenaugh** [reply]
@lenalotti - I am trying to connect an ILI9341 as well to a LILYGO (433mhz) Lora board (Arduino calls it a TTGO Lora32 OLED) with no luck.   Also - when you say an ESP32 DEVKIT V1 - are you saying a regular ESP32 Wroom dev module?  There's no 433 Lora chip on that board so I don't understand how are you able to use it as a balloon chaser.

**Reply by dl9rdz** [reply]
Usually this should work just fine. My standard test board for the TFT displays is a T-Beam. As mentioned above, one critical aspect is the SPI speed, the default (40MHz) is fine for the ILI9225, but the ILI9341 typically requires a lower (20MHz max) speed.

Also, do not use the configuration recommended in the Wiki for the T-Beam 0.7 on any newer T-Beam (v1.0/1.1/1.2). These newer version have the power management chip on some of the pins used in the suggested v0.7 wiring, and having both on the same pins will fail.

**Reply by wrevenaugh** [reply]
Thanks @dl9rdz, but I don't have anything on either the 9341 or the LoraBoard called "RS".   What exactly is that?  "Register Select"?   I can see that being a big source of confusion for people trying to hook up an external display.  And to be clear I have a T3_V1.6.1. Thanks! Bill

**Comment by d2f2** [comment]
Any progress with spi display?

**Reply by dl9rdz** [reply]
Basically it should just be working.

Just saw the old comment on the "RS" pin above: Seems like this is sometimes labelled "DC"
see also http://www.lcdwiki.com/3.2inch_SPI_Module_ILI9341_SKU:MSP3218

And just in case, "SDA", the serial data, is the SDI(MOSI) pin... you can leave the SDO(MISO) pin unconnected.

**Reply by d2f2** [reply]
Hi.
Thank you for your advice! :)
I2c and spi pin names by dif manufactures make me dizzy. Your help is gamę change for me.

As you adviced, I can finally git some data on screen for second. Then goes White.
I had decrased spi speed Then to 10000000 and it works!

Display type 3
Sda 4 - mosi
Clk 15 - sck
Rst 16 - reset
Rs 0 (dont work at 21 - White or gray screen, have to change button from 0 to...) - dc
CS 21 - cs
Spi 10000000

By myself I would be lost for many days. Thank you :)
